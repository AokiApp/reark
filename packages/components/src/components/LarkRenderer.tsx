import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
