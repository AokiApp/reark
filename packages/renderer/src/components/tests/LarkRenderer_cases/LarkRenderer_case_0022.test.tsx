import { renderWithVRT } from "../test-utils/renderWithVRT";
import { LarkRenderer } from "../../LarkRenderer";

describe("LarkRenderer 入れ子構造: 10段ネストのbulletリスト", () => {
  it("10段ネストのbulletリストをレンダリングできる", async () => {
    const blocks = [];
    let parentId = "";
    for (let i = 0; i < 10; i++) {
      const block = {
        block_id: `bullet-nest-${i}`,
        block_type: 12,
        parent_id: parentId,
        bullet: {
          elements: [
            {
              text_run: {
                content: `ネスト${i}`,
                text_element_style: { bold: false },
              },
            },
          ],
          style: { align: "left", folded: false },
        },
        children: i < 9 ? [`bullet-nest-${i + 1}`] : [],
      };
      blocks.push(block);
      parentId = `bullet-nest-${i}`;
    }
    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});
