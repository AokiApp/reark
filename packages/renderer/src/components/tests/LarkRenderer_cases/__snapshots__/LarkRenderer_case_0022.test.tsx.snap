// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 入れ子構造: 10段ネストのbulletリスト 10段ネストのbulletリストをレンダリングできる 2`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-unordered-list"
    >
      <div
        class="reark-text-block"
        style="text-align: left;"
      >
        <span
          class="reark-text"
        >
          ネスト0
        </span>
      </div>
      <div
        class="reark-block"
      >
        <div
          class="reark-unordered-list"
        >
          <div
            class="reark-text-block"
            style="text-align: left;"
          >
            <span
              class="reark-text"
            >
              ネスト1
            </span>
          </div>
          <div
            class="reark-block"
          >
            <div
              class="reark-unordered-list"
            >
              <div
                class="reark-text-block"
                style="text-align: left;"
              >
                <span
                  class="reark-text"
                >
                  ネスト2
                </span>
              </div>
              <div
                class="reark-block"
              >
                <div
                  class="reark-unordered-list"
                >
                  <div
                    class="reark-text-block"
                    style="text-align: left;"
                  >
                    <span
                      class="reark-text"
                    >
                      ネスト3
                    </span>
                  </div>
                  <div
                    class="reark-block"
                  >
                    <div
                      class="reark-unordered-list"
                    >
                      <div
                        class="reark-text-block"
                        style="text-align: left;"
                      >
                        <span
                          class="reark-text"
                        >
                          ネスト4
                        </span>
                      </div>
                      <div
                        class="reark-block"
                      >
                        <div
                          class="reark-unordered-list"
                        >
                          <div
                            class="reark-text-block"
                            style="text-align: left;"
                          >
                            <span
                              class="reark-text"
                            >
                              ネスト5
                            </span>
                          </div>
                          <div
                            class="reark-block"
                          >
                            <div
                              class="reark-unordered-list"
                            >
                              <div
                                class="reark-text-block"
                                style="text-align: left;"
                              >
                                <span
                                  class="reark-text"
                                >
                                  ネスト6
                                </span>
                              </div>
                              <div
                                class="reark-block"
                              >
                                <div
                                  class="reark-unordered-list"
                                >
                                  <div
                                    class="reark-text-block"
                                    style="text-align: left;"
                                  >
                                    <span
                                      class="reark-text"
                                    >
                                      ネスト7
                                    </span>
                                  </div>
                                  <div
                                    class="reark-block"
                                  >
                                    <div
                                      class="reark-unordered-list"
                                    >
                                      <div
                                        class="reark-text-block"
                                        style="text-align: left;"
                                      >
                                        <span
                                          class="reark-text"
                                        >
                                          ネスト8
                                        </span>
                                      </div>
                                      <div
                                        class="reark-block"
                                      >
                                        <div
                                          class="reark-unordered-list"
                                        >
                                          <div
                                            class="reark-text-block"
                                            style="text-align: left;"
                                          >
                                            <span
                                              class="reark-text"
                                            >
                                              ネスト9
                                            </span>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
